cloudify_cli/bootstrap/bootstrap.py

```diff
@@ -514,6 +514,17 @@ def upload_plugin(plugin_path):
                             plugins.validate)
 
     for plugin_source_path in plugin_resources:
+        logger = get_logger()
+        plugin_local_path = \
+            _get_resource_into_dir(temp_dir, plugin_source_path,
+                                   retries, wait_interval, timeout)
+        package = utils.load_plugin_package_details(plugin_local_path)
+        plugin_exists = rest_client.plugins.list(
```

Comment:
maybe make one rest call to read all plugins in the beginning to avoid making a rest call for each plugin resource and then filter on the client side?